showkase-screenshot-testing-paparazzi-sample/src/test/java/com/airbnb/android/showkase/screenshot/testing/paparazzi/sample/MyPaparazziShowkaseScreenshotTest.kt

```diff
@@ -1,9 +1,23 @@
 package com.airbnb.android.showkase.screenshot.testing.paparazzi.sample
 
+import app.cash.paparazzi.DeviceConfig
 import com.airbnb.android.showkase.annotation.ShowkaseScreenshot
+import com.airbnb.android.showkase.screenshot.testing.paparazzi.PaparazziShowkaseDeviceConfig
 import com.airbnb.android.showkase.screenshot.testing.paparazzi.PaparazziShowkaseScreenshotTest
 
 @ShowkaseScreenshot(rootShowkaseClass = PaparazziSampleRootModule::class)
-abstract class MyPaparazziShowkaseScreenshotTest: PaparazziShowkaseScreenshotTest {
-    companion object: PaparazziShowkaseScreenshotTest.CompanionObject
+abstract class MyPaparazziShowkaseScreenshotTest : PaparazziShowkaseScreenshotTest {
+    companion object : PaparazziShowkaseScreenshotTest.CompanionObject {
+        override fun deviceConfigs(): List<PaparazziShowkaseDeviceConfig> {
+            return listOf("en-rGB", "nb-rNO").map { locale ->
```

Comment:
I'm still trying to wrap my head around your use case. If you only want to screenshot test `en-rGB`, why are you also passing the other locale here?

Reply:
@vinaygaba thanks for checking.  

For some previews I want to take one screenshot for each locale and for others I want only one screenshot.
So when setting up the config for the screenshot tests I pass all the locales I want and then I use the group to mark some previews.

**Here's some examples from the app I work on:** 
1. Preview with localised strings. I want one for each of the 5 locales we support to see that the screen looks good in all languages.

Example:
```
@Preview
@Composable
fun NoResultsPreview() {
    NoResultsView(
        imagePainter = painterResource(id = R.drawable.img_no_results_stores),
        text = stringResource(id = R.string.store_search_no_results)
    )
}
```
![image](https://user-images.githubusercontent.com/2194905/233549984-2a3a0584-b5dc-4934-987d-8ffc46fdd455.png)

2. Preview with English dummy string or without texts in it at all. If I still run it for every locale I get 5 screenshots that are exactly the same, so I only want one.

Examples: 
```
@Preview
@Composable
fun FullTopBarPreview() {
    TopBarView(
        title = "Top bar with icons",
        tintColor = White,
        backgroundColor = Orange,
        navigationButton = TopBarNavigationButton.BACK
    ) {
        TopBarIcon(TopBarMenuButton.SEARCH, tint = White)
        TopBarIcon(TopBarMenuButton.SCAN, tint = White)
    }
}
```
![image](https://user-images.githubusercontent.com/2194905/233550217-b2d7e64a-97eb-4681-963c-9aa635147e92.png)

```
@Preview
@Composable
@ShowkaseComposable(group = "notLocalised")
fun QuantityControls99Preview() {
    QuantityControls(quantity = 99)
}
```
![image](https://user-images.githubusercontent.com/2194905/233550233-c8e6f699-1530-4d70-ad3e-fdce3b3c4d75.png)

So my thought is I have to pass all the locales in the device configs and then have logic to skip some tests to avoid 5 screenshots of the same. In my project I do that with the group `notLocalised` and a setup close to Alex Vanyo's and this check: 
```
 // Only take one screenshot if the preview is not localised
 if (locale == "nl-rNL" || componentPreview.showkaseBrowserComponent.group != "notLocalised") {
     paparazzi.snapshot {
         ... 
     }
}
```      

But when I tried moving to your new simpler setup with generated code from https://github.com/airbnb/Showkase/pull/294 I lost the capability to do it this way. 
That's what I am trying to solve with this PR.

Reply:
@andersu The problem is that this change is very specific to your use case and it's not something I'm keen on adding to a library that's meant to be more generic. I think a use case like this is better suited for a custom test class. You can reuse a lot of the code by extending the `PaparazziShowkaseScreenshotTest` class.

Reply:
@vinaygaba I still need access to the `group` when I'm overriding `takePaparazziSnapshot` 
I have updated the PR. Is this something we could add?

Reply:
@andersu Just exposing the group is again limiting as it's specifically for your use case. The lens we want to use is that of something that's more generic. I think exposing the `ShowkaseBrowserComponent` (and color and typography) objects might be more useful. The right solution might be to ensure that they all extend a common sealed class/interface so that we can expose it via the `PaparazziShowkaseTestPreview` interface. What do you think?

Reply:
@vinaygaba I've pushed some changes. Is this what you had in mind?
I'm open to naming suggestions for the sealed interface too. Went with `ShowkaseBrowserElement` for now.